run.sh

```diff
@@ -3,21 +3,103 @@
+    bashio::log.info "Using configured API origin: ${API_ORIGIN}"
+else
+    bashio::log.fatal "api_origin is required! Please configure it in the add-on settings."
+    bashio::log.fatal "Example: http://192.168.1.100:9543"
+    exit 1
+fi
+
+# Check if MQTT service is available (REQUIRED)
+bashio::log.info "Checking for MQTT service..."
+if bashio::services "mqtt" "host" > /dev/null 2>&1; then
+    bashio::log.info "MQTT service IS available from Supervisor"
+    
+    # Extract MQTT credentials from Supervisor services API
+    MQTT_HOST=$(bashio::services "mqtt" "host")
+    export MQTT_HOST
+    MQTT_PORT=$(bashio::services "mqtt" "port")
+    export MQTT_PORT
+    MQTT_USER=$(bashio::services "mqtt" "username")
+    export MQTT_USER
+    MQTT_PASSWORD=$(bashio::services "mqtt" "password")
+    export MQTT_PASSWORD
+    
+    bashio::log.info "MQTT service configured:"
+    bashio::log.info "  Host: ${MQTT_HOST}"
+    bashio::log.info "  Port: ${MQTT_PORT}"
+    bashio::log.info "  User: ${MQTT_USER}"
+else
+    bashio::log.fatal "MQTT service is NOT available!"
+    bashio::log.fatal "This add-on requires the Mosquitto broker add-on."
+    bashio::log.fatal "Please install and start the Mosquitto broker add-on first."
+    bashio::log.fatal ""
+    bashio::log.fatal "Installation steps:"
+    bashio::log.fatal "  1. Go to Settings > Add-ons > Add-on Store"
+    bashio::log.fatal "  2. Search for 'Mosquitto broker'"
+    bashio::log.fatal "  3. Install the official Mosquitto broker add-on"
+    bashio::log.fatal "  4. Start the Mosquitto broker"
+    bashio::log.fatal "  5. Restart this add-on"
+    exit 1
+fi
+
+# Set environment variables for Node.js
+export API_ORIGIN
+export PROXY_PORT
+export CONTROL_PORT
+export INGRESS_PORT
+export ENTRY_KEY_TTL_SECONDS
+export DEBUG_LOGGING
+export DEBUG_LOGS_DIR=/data/debug-logs
 export SQLITE3_ENABLED=true
 export SQLITE3_DB_PATH=/data/database.sqlite
 
-bashio::log.info "Environment configured:"
-bashio::log.info "  - PROXY_PORT: ${PROXY_PORT}"
-bashio::log.info "  - CONTROL_PORT: ${CONTROL_PORT}"
-bashio::log.info "  - SQLITE3_DB_PATH: ${SQLITE3_DB_PATH}"
+bashio::log.info "Configuration:"
+bashio::log.info "  API_ORIGIN: ${API_ORIGIN}"
+bashio::log.info "  PROXY_PORT: ${PROXY_PORT} (container listen port)"
+bashio::log.info "  CONTROL_PORT: ${CONTROL_PORT}"
+bashio::log.info "  INGRESS_PORT: ${INGRESS_PORT}"
+bashio::log.info "  DEBUG_LOGGING: ${DEBUG_LOGGING}"
+bashio::log.info "  MQTT_HOST: ${MQTT_HOST}"
+bashio::log.info "  MQTT_PORT: ${MQTT_PORT}"
+bashio::log.info ""
+bashio::log.info "Nest devices will connect to: ${API_ORIGIN}"
+
+# Start the vendor API server FIRST (creates database tables)
+bashio::log.info "Starting API server..."
+cd /server || exit 1
+node dist/index.js &
+SERVER_PID=$!
+bashio::log.info "API server started (PID: ${SERVER_PID})"
+
+# Wait for server to initialize database
+bashio::log.info "Waiting for database initialization..."
+sleep 3
```

Comment:
just checking if you think 3 seconds will be fine on any system?
otherwise I might suggest polling with timeout:
```suggestion
MAX_WAIT=30
WAIT_COUNT=0
while [ $WAIT_COUNT -lt $MAX_WAIT ]; do
  if [ -f /data/database.sqlite ]; then
    bashio::log.info "Database ready"
    break
  fi
  sleep 1
  WAIT_COUNT=$((WAIT_COUNT + 1))
done

if [ $WAIT_COUNT -eq $MAX_WAIT ]; then
  bashio::log.fatal "Database initialization timeout after ${MAX_WAIT} seconds"
  kill $SERVER_PID 2>/dev/null
  exit 1
fi
```